### PR TITLE
Decouple anchors from titles

### PR DESCRIPTION
**Pull request by heymatthew**
Fixes anchors on OIDC flow page, and decouples anchors in other places too.

### CODE REVIEW

File: src/auth.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ website (such as "https://yourapp.example.com/oidc-callback") or mobile app
 
 Your application can use any OIDC client to negotiate the authentication flow
 but it must support OIDC authorization code flow with [PKCE][pkce]{:.external}.
-See the [Example Web Browser OIDC Consumer][oidcjs-example] section below for a
+See the [Example Web Browser OIDC Consumer](#oidcjs-example-anchor) section below for a
```
Comment (heymatthew) [inline src/auth.md:66]: This link is currently broken in the wild.

Comment (ncjones) [reply src/auth.md:66]: I don't like the anchor suffix. It's redundant.

Comment (heymatthew) [reply src/auth.md]: Using a suffix that'll never show up in a title ensures we wont end up generating multiple IDs on the page. The thinking is that someone is unlikely to put the word 'anchor' in their heading, but it's likely that a heading may change without updating the dom.

I've gone and removed the `-anchor` suffix. Lets just try to keep the anchor IDs different to what we generate with titles and it should all be fine 👌

Comment (ncjones) [reply src/auth.md]: We should auto generate anchors from section headings and the linter should detect broken links.

**Review by ncjones — APPROVED** [review]

File: src/auth.md (hunk 1 above)
Comment (heymatthew) [reply src/auth.md]: +1, lets investigate a linter in a followup PR and remove this when we get there.

**Pull request merged**